### ISSUE TITLE
Implement an "authenticate_user" helper that doen't fail

### ISSUE THREAD

**Issue by cmolina** [mcordell/grape_devise_token_auth#3]
It could be very useful if I can make controllers return information customized to a user if he/she is logged in, otherwise return default information.

Right now I must rescue the error by doing something like this:

``` ruby
begin
  user = authenticate_user!
  info = user.some_specific_info
rescue => error
  info = generic_info
ensure
  present info
end
```

I'd love to have a helper that doesn't fail (nor throw 400 error), so the following code will be equivalent:

``` ruby
user = authenticate_user     # there is no "!"
if user
  info = user.some_specific_info
else
  info = generic_info
end
present info
```

Thank you in advance.

**Comment by cmolina** [comment]
@mcordell would you accept a PR with a proposal?

**Comment by mcordell** [comment]
I think that is a solid idea and would definitely look at a PR

**Comment by cmolina** [comment]
There it is. I also edited the README.md with a small description.

**Comment by mcordell** [comment]
Closed in #4
